src/app/core/store/store.service.ts

```diff
@@ -12,4 +12,15 @@ export class StoreService {
   getProductListing(): Observable<Array<Product>> {
     return this.http.get<Array<Product>>(`${environment.API_URL}/store/products`);
   }
+
+  getProduct(index: number): Observable<Product> {
+    return this.http.get<Product>(`${environment.API_URL}/store/products/` + index);
+  }
+
+  addToCart(productId: number, size: string, quantity: number) {
```

Comment:
instead of passing all of these params, I would have a addToCartModel that gets passed in.